### ISSUE TITLE
[JSDoc/checkjs]: exported @typedef doesn't work with mapped types directly

### ISSUE THREAD

**Issue by Hotell** [Microsoft/TypeScript#24916]
**TypeScript Version:**  2.9.1

**Search Terms:** 
- checkJS
- JSDoc

**Expected behavior:**
following should work without errors:

- *types.js*

```js
// @ts-check
export {}

/**
 * @typedef {Partial<{one:boolean} & {two:string}>} Test
 */
```

- *index.js*

```js
// @ts-check
/**
 * @type {import('./types').Test}
 */
const test = {}
```

**Actual behavior:**

- *types.js*

```js
// @ts-check
export {}

/**
 * @typedef {Partial<{one:boolean} & {two:string}>} Test
 */
```

- *index.js*

```js
// @ts-check
/**
 * @type {import('./types').Test}
 */
const test = {}
```

exported `@typedef` definitions, which are using Mapped types transformations (`Partial`,...), will error out with cryptic error when they are imported  via `/** @type {import('./types).MappedTest}*` to annotate a variable :

![image](https://user-images.githubusercontent.com/1223799/41321413-1bf9b11e-6ea4-11e8-87ee-e546a96205c6.png)

Although creating a variable which is typed via mapped type and exported as a token value works:

- *types.js*

```js
// @ts-check
export {}

/**
 * @typedef {Partial<{one:boolean} & {two:string}>} Test
 */

/**
 * @type {Test}
 */
export var Test
```

- *index.js*

```js
// @ts-check
/**
 * @type {typeof import('./types').Test}
 */
const test = {}
```

![image](https://user-images.githubusercontent.com/1223799/41321780-bec5f92e-6ea5-11e8-93dd-70f33f2e7cc0.png)

**Comment by weswigham** [comment]
@sandersn This seems to be fixed in `master`, so I assume one of your many refactorings to jsdoc handling in the last two weeks have squelched this bug. Should we add a regression test, or do you know if one of the tests you've added covers this scenario?

**Comment by sandersn** [comment]
Our test coverage is pretty bad right now, so I would add a test. In particular, I didn’t test typedef with Es2015 modules, that I can remember at least.

**Comment by Hotell** [comment]
Just to mention, It's not with es2015 modules , just type import via `import()`

**Comment by sandersn** [comment]
Yes, but it's imported from an es2015 module, which might have been the cause of the assert.

**Comment by RyanCavanaugh** [comment]
@Hotell just to clarify, is this fixed, or not?

**Comment by Hotell** [comment]
looks fixed with 3.1.1

![image](https://user-images.githubusercontent.com/1223799/46766525-26c23a00-ccda-11e8-8884-a1ac900eb55b.png)


Cheers !
